test/integration/certificates/duration_test.go

```diff
@@ -51,12 +51,12 @@ import (
 func TestCSRDuration(t *testing.T) {
 	t.Parallel()
 
-	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
-	t.Cleanup(cancel)
-
 	s := kubeapiservertesting.StartTestServerOrDie(t, nil, nil, framework.SharedEtcd())
 	t.Cleanup(s.TearDownFn)
 
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	t.Cleanup(cancel)
```

Comment:
so many t.Cleanup() functions is not complex to follow?

Reply:
somewhat - I was just trying to do minimal changes...